prow/github/types.go

```diff
@@ -1090,14 +1101,70 @@ type Milestone struct {
 // Note that it's wrapping a GitCommit, so author/committer information is in two places,
 // but contain different details about them: in RepositoryCommit "github details", in GitCommit - "git details".
 type RepositoryCommit struct {
-	SHA         string    `json:"sha"`
-	Commit      GitCommit `json:"commit"`
-	Author      User      `json:"author"`
-	Committer   User      `json:"committer"`
-	Parents     []Commit  `json:"parents,omitempty"`
-	HTMLURL     string    `json:"html_url"`
-	URL         string    `json:"url"`
-	CommentsURL string    `json:"comments_url"`
+	NodeID      string   `json:"node_id,omitempty"`
+	SHA         string   `json:"sha,omitempty"`
+	Commit      Commit   `json:"commit,omitempty"`
+	Author      User     `json:"author,omitempty"`
+	Committer   User     `json:"committer,omitempty"`
+	Parents     []Commit `json:"parents,omitempty"`
+	HTMLURL     string   `json:"html_url,omitempty"`
+	URL         string   `json:"url,omitempty"`
+	CommentsURL string   `json:"comments_url,omitempty"`
+
+	// Details about how many changes were made in this commit. Only filled in during GetCommit!
+	Stats CommitStats `json:"stats,omitempty"`
+	// Details about which files, and how this commit touched. Only filled in during GetCommit!
+	Files []CommitFile `json:"files,omitempty"`
+}
+
+// CommitStats represents the number of additions / deletions from a file in a given RepositoryCommit or GistCommit.
+type CommitStats struct {
+	Additions int `json:"additions,omitempty"`
+	Deletions int `json:"deletions,omitempty"`
+	Total     int `json:"total,omitempty"`
+}
+
+// CommitFile represents a file modified in a commit.
+type CommitFile struct {
+	SHA              string `json:"sha,omitempty"`
+	Filename         string `json:"filename,omitempty"`
+	Additions        int    `json:"additions,omitempty"`
+	Deletions        int    `json:"deletions,omitempty"`
+	Changes          int    `json:"changes,omitempty"`
+	Status           string `json:"status,omitempty"`
+	Patch            string `json:"patch,omitempty"`
+	BlobURL          string `json:"blob_url,omitempty"`
+	RawURL           string `json:"raw_url,omitempty"`
+	ContentsURL      string `json:"contents_url,omitempty"`
+	PreviousFilename string `json:"previous_filename,omitempty"`
+}
+
+// Commit represents a GitHub commit.
+type Commit struct {
+	SHA          string                `json:"sha,omitempty"`
+	Author       User                  `json:"author,omitempty"`
+	Committer    User                  `json:"committer,omitempty"`
+	Message      string                `json:"message,omitempty"`
+	Tree         Tree                  `json:"tree,omitempty"`
+	Parents      []Commit              `json:"parents,omitempty"`
+	Stats        CommitStats           `json:"stats,omitempty"`
+	HTMLURL      string                `json:"html_url,omitempty"`
+	URL          string                `json:"url,omitempty"`
+	Verification SignatureVerification `json:"verification,omitempty"`
```

Comment:
Please make this a pointer, because it might be empty

Reply:
Fixed.